backend/src/apiserver/server/list_request_util.go

```diff
@@ -157,3 +158,27 @@ func deserializePageToken(pageToken string) (*common.Token, error) {
 	}
 	return &token, nil
 }
+
+// parseAPIFilter attempts to decode a base-64 encoded JSON-stringified api
+// filter object. An empty string is considered valid input, and equivalent to
+// the nil filter, which trivially does nothing.
+func parseAPIFilter(encoded string) (*api.Filter, error) {
+	if encoded == "" {
+		return nil, nil
+	}
+
+	errorF := func(err error) (*api.Filter, error) {
```

Comment:
is any good reason of having this function instead of returning errors directly?

Reply:
Avoids duplicate in error handling, since the message is the same. It's a very idiomatic Go thing to do.